src/agreement/mod.rs

```diff
@@ -121,12 +122,32 @@ impl AgreementContent {
 }
 
 /// Messages sent during the binary Byzantine agreement stage.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Rand)]
 pub struct AgreementMessage {
     pub epoch: u32,
     pub content: AgreementContent,
 }
 
+// note: extending rand_derive to correct generate random values from boxes would make this
+// implementation obsolete; however at the time of this writing, `rand::Rand` is already deprecated
+// with no replacement in sight.
+impl rand::Rand for AgreementContent {
+    fn rand<R: rand::Rng>(rng: &mut R) -> Self {
+        let message_type = *rng
+            .choose(&["bval", "aux", "conf", "term", "coin"])
+            .unwrap();
+
+        match message_type {
+            "bval" => AgreementContent::BVal(rand::random()),
+            "aux" => AgreementContent::Aux(rand::random()),
+            "conf" => AgreementContent::Conf(rand::random()),
+            "term" => AgreementContent::Conf(rand::random()),
```

Comment:
Could we use the pattern matching "or" here? 

```rust
"conf" | "term" => AgreementContent::Conf(rand::random()),
```

Reply:
Ah, good catch! That's probably supposed to say `AgreementContent::Term`?

Reply:
Fixed.